### PR TITLE
C# cleanup client-side call RECV_STATUS_ON_CLIENT internals

### PR DESCRIPTION
**Pull request by jtattermusch**
-- a bit of cleanup in AsyncCall
-- added some tests

### CODE REVIEW

File: src/csharp/Grpc.Core/Internal/AsyncCall.cs [hunk 1]
```diff
@@ -407,12 +407,15 @@ private void HandleReceivedResponseHeaders(bool success, Metadata responseHeader
         /// <summary>
         /// Handler for unary response completion.
         /// </summary>
-        private void HandleUnaryResponse(bool success, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
+        private void HandleUnaryResponse(bool sucess, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
```
Comment (murgatroid99) [inline src/csharp/Grpc.Core/Internal/AsyncCall.cs]: This change introduced a misspelling. `success` was right the first time.

File: src/csharp/Grpc.Core/Internal/AsyncCall.cs [hunk 2]
```diff
@@ -407,12 +407,15 @@ private void HandleReceivedResponseHeaders(bool success, Metadata responseHeader
...
         {
+            // NOTE: because this event is a result of batch containing GRPC_OP_RECV_STATUS_ON_CLIENT,
+            // success will be always set to true.
```
Comment (murgatroid99) [inline src/csharp/Grpc.Core/Internal/AsyncCall.cs]: Are you sure about this? Does the batch *only* contain a `GRPC_OP_RECV_STATUS_ON_CLIENT` op?

Comment (jtattermusch) [inline src/csharp/Grpc.Core/Internal/AsyncCall.cs]: The recv_status_on_client will also make the other ops succeed.
https://github.com/grpc/grpc/blob/master/include/grpc/impl/codegen/grpc_types.h#L307
* This operation always succeeds, meaning ops paired with this operation
      will also appear to succeed, even though they may not have. In that case
      the status will indicate some failure.
*

File: src/csharp/Grpc.Core/Internal/AsyncCall.cs (hunk 1 above)
Comment (jtattermusch) [inline src/csharp/Grpc.Core/Internal/AsyncCall.cs]: Fixed.

**Comment by murgatroid99** [comment]
LGTM

**Pull request merged**